tls/s2n_resume.c

```diff
@@ -311,7 +312,11 @@ int s2n_connection_is_session_resumed(struct s2n_connection *conn)
 int s2n_connection_is_ocsp_stapled(struct s2n_connection *conn)
 {
     notnull_check(conn);
-    return IS_OCSP_STAPLED(conn->handshake.handshake_type) ? 1 : 0;
+
+    return
+        conn->actual_protocol_version >= S2N_TLS13 ?
+            (s2n_server_can_send_ocsp(conn) || s2n_server_sent_ocsp(conn)) :
+            IS_OCSP_STAPLED(conn->handshake.handshake_type) ? 1 : 0;
```

Comment:
Having doubly nested ternary operators is hard to read, and having a macro `IS_OCSP_STAPLED()` that doesn't return `0` or `1` is confusing. I think we should fix both.

I think we should change this section to:
```
if (conn->actual_protocol_version >= S2N_TLS13) {
    return (s2n_server_can_send_ocsp(conn) || s2n_server_sent_ocsp(conn));
} else {
    return IS_OCSP_STAPLED(conn->handshake.handshake_type);
}
```

And change the [`IS_OCSP_STAPLED`](https://github.com/awslabs/s2n/blob/4c81b4b13c86780ab085e04beff0c2ab8e2f6fac/tls/s2n_handshake.h#L167) macro definition to:
```
#define IS_OCSP_STAPLED( type ) ( ((type) & OCSP_STATUS) != 0 )
```